apps/desktop/tests/services/AgentService/execution/executor.test.ts

```
@@ -0,0 +1,90 @@
+}
+
+describe('AiRequestExecutor', () => {
+    beforeEach(() => {
+        vi.clearAllMocks();
+    });
+
+    it('keeps the original request error when provider error classification fails', async () => {
+        const requestError = new Error(
+            'error sending request for url (https://hub.touch-ai.org/api/v1/chat/completions)'
+        );
+        providerMock.stream.mockImplementation(async function* () {
```

ESLint

[error] This generator function does not have 'yield'.

(require-yield)